### PR TITLE
Add Fusion support to Azure batch executor

### PR DESCRIPTION
**Pull request by bentsherman**
Fusion support for the azure batch task handler is trickier because much of the task logic (container options, environment vars, submit command) are handled in a separate class `AzBatchService`. So we either need to pass the executor and script launcher all the way up the call stack or move some of the fusion methods into `FusionHelper`.

I'm currently trying the latter approach to see how it works. So `AzBatchTaskHandler` doesn't implement the `FusionAwareTask` trait at all, it just calls helper functions from `FusionHelper`.

Still need to update unit tests and test on azure.

### CODE REVIEW

File: modules/nextflow/src/main/groovy/nextflow/executor/fusion/FusionHelper.groovy [hunk 1]
```diff
@@ -39,6 +40,17 @@ class FusionHelper {
         return result!=null ? result.toString()=='true' : false
     }
 
+    @Memoized
+    static FusionScriptLauncher getFusionLauncher(TaskRun task) {
+        FusionScriptLauncher.create(task.toTaskBean(), task.workDir.scheme)
+    }
```
Comment (bentsherman) [inline modules/nextflow/src/main/groovy/nextflow/executor/fusion/FusionHelper.groovy:46]: I'm not sure if this is really clever or really stupid, but it was a convenient way to access the fusion launcher for `AzBatchService`.

**Comment by pditommaso** [comment]
Closing this until the support for Azure blob store is implemented in the Fusion driver

File: modules/nextflow/src/main/groovy/nextflow/executor/fusion/FusionHelper.groovy (hunk 1 above)
Comment (pditommaso) [reply modules/nextflow/src/main/groovy/nextflow/executor/fusion/FusionHelper.groovy]: yeah, this was tricky. Made some refactors to simplify it

**Comment by pditommaso** [comment]
@pabloaledo should be in a better shape now

**Comment by pabloaledo** [comment]
Thanks !!

**Comment by pditommaso** [comment]
@bentsherman how did you manage to make other PR to run? here tests are skipped

**Comment by bentsherman** [comment]
Probably the slash in the branch name 😄

**Comment by pditommaso** [comment]
Yeah, but here there were new updates

**Comment by bentsherman** [comment]
I just renamed the branch and made a new PR, but now the tests ran for this one 🤷

File: modules/nf-commons/src/main/nextflow/Global.groovy [hunk 2]
```diff
@@ -165,6 +165,17 @@ class Global {
...
+
+        String a
+
+        if( env && (a=env.AZURE_STORAGE_ACCOUNT_KEY) )  {
+            log.debug "Using AZ credentials defined by environment variables AZURE_STORAGE_ACCOUNT_KEY"
+            return [a]
+        }
+        return null
+    }
+
```
Comment (pditommaso) [inline modules/nf-commons/src/main/nextflow/Global.groovy:178]: This is an ugly legacy pattern that's used for the AWS credentials I'm trying to get rid in upcoming version. Let's not do the same for Azure.

Azure configuration can be retrieved via [AzConfig.config](https://github.com/nextflow-io/nextflow/blob/025ff9d0cc9d67a923896d58f89d8b2ac25defa1/plugins/nf-azure/src/main/nextflow/cloud/azure/config/AzConfig.groovy#L71-L73)

File: modules/nf-commons/src/main/nextflow/Global.groovy [hunk 3]
```diff
@@ -200,10 +211,18 @@ class Global {
         getAwsCredentials(env, config)
     }
 
+    static List<String> getAzCredentials(Map env) {
+        getAzCredentials(env, config)
+    }
+
     static List<String> getAwsCredentials() {
```
Comment (pditommaso) [inline modules/nf-commons/src/main/nextflow/Global.groovy:218]: Same as above

File: plugins/nf-azure/src/main/nextflow/cloud/azure/fusion/AzFusionEnv.groovy [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ */
+@Extension
+@CompileStatic
+class AzFusionEnv implements FusionEnv {
+
+    @Override
+    Map<String, String> getEnvironment(String scheme, FusionConfig config) {
+
+        final result = new HashMap<String,String>()
+        final creds = config.exportAzAccessKeys() ? Global.getAzCredentials() : Collections.<String>emptyList()
```
Comment (pditommaso) [inline plugins/nf-azure/src/main/nextflow/cloud/azure/fusion/AzFusionEnv.groovy:40]: Use [AzConfig](https://github.com/nextflow-io/nextflow/blob/025ff9d0cc9d67a923896d58f89d8b2ac25defa1/plugins/nf-azure/src/main/nextflow/cloud/azure/config/AzConfig.groovy#L71-L73) instead

File: modules/nf-commons/src/main/nextflow/Global.groovy (hunk 2 above)
Comment (pditommaso) [reply modules/nf-commons/src/main/nextflow/Global.groovy:178]: @pablo-aledo there are couple of comments to address

**Comment by pditommaso** [comment]
Let's merge this!

**Review by pditommaso — APPROVED** [review]

**Pull request merged**